zebra-state/src/service/finalized_state/zebra_db.rs

```diff
@@ -106,6 +107,14 @@ impl ZebraDb {
         )
         .expect("unable to read database format version file");
 
+        DiskDb::try_reusing_previous_db_after_major_upgrade(
```

Comment:
It would be nice to:
- apply any required format upgrades before returning the db when the code format version is a higher major version than the disk format version, 
- apply any format upgrades for the previous major version at the old path, and,
- apply the major format upgrade in a temp directory, then move it to the new db path once it's been upgraded.

Reply:
This sounds like it would make opening the db much slower than it is. Currently, opening the db is fast because db upgrades run concurrently, and moving the db is instant because Zebra just renames the path.

I'm now thinking we should delete all previous db upgrades, and keep only the framework for doing them. The reason is that we don't need them anymore as there aren't and won't be any Zebra instances that could use them. If we ever need another DB upgrade, we can either:

- transfer the old database and use the framework for upgrades, or
- apply the framework for upgrades to the current db version as we did in the past, or
- start syncing from scratch.

Reply:
Note that it is tricky to use temp directories because they can be in another device and the move won't work, unless we make sure we create the temp directory alongisde the db folders (which may have unintended consequences?)